projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,30 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+)
+
+func Execute() {
+	arg := "."
+	if len(os.Args) > 1 {
+		arg = os.Args[1]
+	}
+	if arg == "-h" {
+		fmt.Println("go-ls : list directory contents")
+	} else {
+		entries, err := os.ReadDir(arg)
+		if err != nil {
+			fdopendir := fmt.Sprintf("fdopendir %s: not a directory", arg)
+			if err.Error() == fdopendir {
```

Comment:
I like the way you check for specific errors here, but in golang there is a better way. You can use [errors.Is](https://pkg.go.dev/errors#example-Is) and you can use the error that's already defined in go for "not a directory"

Reply:
I couldn't find any predefined error for `not a directory` in go ,  the only ones that I can find : 
on the `os package` are : 
```
var (
	// ErrInvalid indicates an invalid argument.
	// Methods on File will return this error when the receiver is nil.
	ErrInvalid = fs.ErrInvalid // "invalid argument"

	ErrPermission = fs.ErrPermission // "permission denied"
	ErrExist      = fs.ErrExist      // "file already exists"
	ErrNotExist   = fs.ErrNotExist   // "file does not exist"
	ErrClosed     = fs.ErrClosed     // "file already closed"

	ErrNoDeadline       = errNoDeadline()       // "file type does not support deadline"
	ErrDeadlineExceeded = errDeadlineExceeded() // "i/o timeout"
)
```
Is there any other package using it ?

Reply:
I think that specific error comes from `syscall.ENOTDIR`